### PR TITLE
Add JsonAdapter class checker + tests

### LINKED ISSUE #23: Add `@JsonAdapter` annotation checker
Similar to Moshi's `@JsonClass` checker, but for GSON

```kotlin
/** Checks for classes annotated with GSON's [JsonAdapter] annotation. */
private class JsonAdapterAnnotationChecker : ClassChecker {
  override fun serializerFor(rawType: Class<*>): Serializer? {
    if (rawType.isAnnotationPresent(JsonAdapter::class.java)) {
      return Serializer.GSON
    }
    return null
  }
}
```

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #23 with some beefier tests

### CODE REVIEW

File: src/main/kotlin/com/slack/moshi/interop/gson/ClassCheckers.kt [hunk 1]
```diff
@@ -52,6 +53,16 @@ public object JsonClassClassChecker : ClassChecker {
   override fun toString(): String = "JsonClassClassChecker"
 }
 
+/** Checks for classes annotated with GSON's [JsonAdapter] annotation. */
+public object JsonAdapterAnnotationChecker : ClassChecker {
```
Comment (adamsp) [inline src/main/kotlin/com/slack/moshi/interop/gson/ClassCheckers.kt:57]: Missing the `toString()` override.

Comment (adamsp) [inline src/main/kotlin/com/slack/moshi/interop/gson/ClassCheckers.kt:57]: We have inconsistent naming between `JsonAdapterAnnotationChecker` and `JsonClassClassChecker`. Either we rename this `JsonAdapterClassChecker` or we rename them both to `JsonAdapterAnnotationClassChecker` format. I prefer the second one actually - it's clear we're checking for an annotation, but also consistent with other `ClassChecker` naming.

**Review by adamsp — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/slack/moshi/interop/gson/ClassCheckers.kt (hunk 1 above)
Comment (ZacSweers) [reply src/main/kotlin/com/slack/moshi/interop/gson/ClassCheckers.kt]: 363988a

Comment (ZacSweers) [reply src/main/kotlin/com/slack/moshi/interop/gson/ClassCheckers.kt]: 363988a

**Review by adamsp — APPROVED** [review]

**Pull request merged**